Resources/public/js/views/services/ez-locationviewviewservice.js

```diff
@@ -208,6 +209,42 @@ YUI.add('ez-locationviewviewservice', function (Y) {
         },
 
         /**
+         * translate event handler, makes the application to navigate to edit content available
+         * in the facade with given language and base language
+         *
+         * @method _translateContent
+         * @protected
+         * @param {EventFacade} e
+         * @param {String} e.languageCode language code of new translation
+         * @param {Boolean} e.isBaseLanguage defines if new translation is basing on another one or not
+         *                  if it's set to true then languageCode of base language will be set, otherwise
+         *                  base language will be ommited
+         * @param {String} e.baseLanguageCode language code of existing translation on which 
+         *                  new translation will be basing, it it's not set then current
+         *                  languageCode will be used
+         */
+        _translateContent: function (e) {
```

Comment:
> 4. When calling confirmHandler in languageSelectionBoxView, I'm calling the `translate` event which in consequence is redirecting to edit content view.

[...]

> In the edit content view there are few issues I have faced to:
4.1. currently when I pass languageCode of not existing translation (new translation) fields in content edit view are filled with data from main language version - shouldn't they be empty (for now at least when we ommit the baseLanguage)?
4.2. when saving/publishing the new translation I receives 404 from api call with message "Could not find 'Language' with identifier 'ger-DE'", which probably points on that the version for this language wasn't created. Is it bug or maybe something I should develop?

it's because in this case you should use the create content route (and most likely you'll have to tweak its languageCode handling like you did for the edit so that it's not always the main language code)

> 4.3. there should be definition of edit content view changed in router but when I was playing with it a bit it just made me frustrated :) because I wasn't able to set the parameter in URI to be optional. I suppose that for translations should be written new routes with dividing them to creating new translation without basing on any other translation and another one with passing base translation. Is my way of thinking correct or I'm wrong?

indeed you can not have optional parameters in route but you can *duplicate* the route, something like:
```js
                {
                    name: "editContent",
                    path: '/edit/:id/:languageCode',
                    service: Y.eZ.ContentEditViewService,
                    sideViews: {'navigationHub': false, 'discoveryBar': false},
                    view: 'contentEditView',
                    callbacks: ['open', 'checkUser', 'handleSideViews', 'handleMainView']
                },
                {
                    name: "editContentBaseLanguage",
                    path: '/edit/:id/:languageCode/:baseLanguageCode',
                    service: Y.eZ.ContentEditViewService,
                    sideViews: {'navigationHub': false, 'discoveryBar': false},
                    view: 'contentEditView',
                    callbacks: ['open', 'checkUser', 'handleSideViews', 'handleMainView']
                },
```